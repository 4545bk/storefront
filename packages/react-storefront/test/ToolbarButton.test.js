/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import ToolbarButton from '../src/ToolbarButton'

describe('ToolbarButton', () => {
  it('should render icon and button', () => {
    expect(
      mount(
        <ToolbarButton icon={<div>icon here</div>} label="menu"/>
      )
    ).toMatchSnapshot()
  })

  it('should render additional children inside the button', () => {
    expect(
      mount(
        <ToolbarButton icon={<div>icon here</div>} label="menu">
          <div>additional child</div>
        </ToolbarButton>
      )
    ).toMatchSnapshot()
  })
})