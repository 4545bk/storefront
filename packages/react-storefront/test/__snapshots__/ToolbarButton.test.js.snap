// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarButton should render additional children inside the button 1`] = `
<WithStyles(ToolbarButton) icon={{...}} label="menu">
  <ToolbarButton classes={{...}} icon={{...}} label="menu">
    <WithStyles(IconButton) classes={{...}}>
      <IconButton classes={{...}} color="default" disabled={false}>
        <WithStyles(ButtonBase) className="MuiIconButton-root-2" centerRipple={true} focusRipple={true} disabled={false}>
          <ButtonBase classes={{...}} className="MuiIconButton-root-2" centerRipple={true} focusRipple={true} disabled={false} disableRipple={false} tabIndex="0" type="button">
            <button onBlur={[Function: handleEvent]} onFocus={[Function: value]} onKeyDown={[Function: value]} onKeyUp={[Function: value]} onMouseDown={[Function: handleEvent]} onMouseLeave={[Function: handleEvent]} onMouseUp={[Function: handleEvent]} onTouchEnd={[Function: handleEvent]} onTouchMove={[Function: handleEvent]} onTouchStart={[Function: handleEvent]} tabIndex="0" className="MuiButtonBase-root-8 MuiIconButton-root-2" type="button" disabled={false}>
              <span className="MuiIconButton-label-7">
                <div className="RSFToolbarButton-wrap-1">
                  <div>
                    icon here
                  </div>
                  <div>
                    menu
                  </div>
                </div>
                <div>
                  additional child
                </div>
              </span>
              <WithStyles(TouchRipple) innerRef={[Function: value]} center={true}>
                <TouchRipple classes={{...}} center={true}>
                  <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-11" childFactory={[Function: childFactory]}>
                    <span className="MuiTouchRipple-root-11" />
                  </TransitionGroup>
                </TouchRipple>
              </WithStyles(TouchRipple)>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </IconButton>
    </WithStyles(IconButton)>
  </ToolbarButton>
</WithStyles(ToolbarButton)>
`;

exports[`ToolbarButton should render icon and button 1`] = `
<WithStyles(ToolbarButton) icon={{...}} label="menu">
  <ToolbarButton classes={{...}} icon={{...}} label="menu">
    <WithStyles(IconButton) classes={{...}}>
      <IconButton classes={{...}} color="default" disabled={false}>
        <WithStyles(ButtonBase) className="MuiIconButton-root-2" centerRipple={true} focusRipple={true} disabled={false}>
          <ButtonBase classes={{...}} className="MuiIconButton-root-2" centerRipple={true} focusRipple={true} disabled={false} disableRipple={false} tabIndex="0" type="button">
            <button onBlur={[Function: handleEvent]} onFocus={[Function: value]} onKeyDown={[Function: value]} onKeyUp={[Function: value]} onMouseDown={[Function: handleEvent]} onMouseLeave={[Function: handleEvent]} onMouseUp={[Function: handleEvent]} onTouchEnd={[Function: handleEvent]} onTouchMove={[Function: handleEvent]} onTouchStart={[Function: handleEvent]} tabIndex="0" className="MuiButtonBase-root-8 MuiIconButton-root-2" type="button" disabled={false}>
              <span className="MuiIconButton-label-7">
                <div className="RSFToolbarButton-wrap-1">
                  <div>
                    icon here
                  </div>
                  <div>
                    menu
                  </div>
                </div>
              </span>
              <WithStyles(TouchRipple) innerRef={[Function: value]} center={true}>
                <TouchRipple classes={{...}} center={true}>
                  <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-11" childFactory={[Function: childFactory]}>
                    <span className="MuiTouchRipple-root-11" />
                  </TransitionGroup>
                </TouchRipple>
              </WithStyles(TouchRipple)>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </IconButton>
    </WithStyles(IconButton)>
  </ToolbarButton>
</WithStyles(ToolbarButton)>
`;
